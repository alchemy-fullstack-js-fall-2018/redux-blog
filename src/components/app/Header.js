import React from 'react';
import { navLinks } from '../../routes';

export default function Header() {
  return (
    <header>
      <nav>
        {navLinks}
      </nav>
      <h1>Redux Blog</h1>
    </header>
  );
}
