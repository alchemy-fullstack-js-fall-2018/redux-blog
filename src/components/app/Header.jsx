import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const StyledHeader = styled.header`
  background-color: ${ ({ theme }) => theme.primary };
  color: ${ ({ theme }) => theme.secondary };

  h1 {
    font-family: BOMBARD, "Avenir Next", sans-serif;
    font-weight: bolder;
    letter-spacing: 2px;
    font-size: 36px;
  }
`;

const StyledNav = styled.nav`
  margin: 0px 0px 15px 0px;
`;

const StyledLink = styled(Link)`
  box-sizing: border - box;

  text-decoration: none;
  color: ${ ({ theme }) => theme.secondary };
  margin: 5px;
  padding: 8px;
  border: 1px solid white;

  &:hover {
    color: ${ ({ theme }) => theme.darksecondary };
    border: 1px solid ${ ({ theme }) => theme.darksecondary };
  }
`;

const Header = () => {

  const links = [
    { label: 'Home', path: '/home' },
    { label: 'Posts', path: '/posts/search' },
    { label: 'Users', path: '/users' },
    { label: 'Register', path: '/users/new' }
  ];
  const LinkComponents = links.map(link => {
    return (
      <StyledLink key={link.label} to={link.path}>
        {link.label}
      </StyledLink>
    );
  });

  return (
    <StyledHeader>
      <h1>Redux Blogging</h1>
      <StyledNav>
        {LinkComponents}
      </StyledNav>
    </StyledHeader>
  );
};

export default Header;
