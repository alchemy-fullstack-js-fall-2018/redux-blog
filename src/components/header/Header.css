#header {
  text-align: center;
  background-color: lightgreen;
  padding: 25px;

  h1 {
    padding-bottom: 10px;
    margin: 0;
  }

  nav {
    display: flex;
    justify-content: space-around;
  }
}