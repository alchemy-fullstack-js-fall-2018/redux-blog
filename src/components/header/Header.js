import { ROUTES } from '../../routes/routes';
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Header.css';

export default function Header() {
  return (
    <div id={styles.header}>
      <h1>BLOGGY McBLOG BLOG!</h1>
      <nav>
        <Link to={ROUTES.HOME.linkTo()}>Home</Link>
        <Link to={ROUTES.USERS.linkTo()}>Users</Link>
      </nav>
    </div>
  );
}
