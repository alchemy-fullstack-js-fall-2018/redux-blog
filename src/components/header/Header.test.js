import { shallow } from 'enzyme';
import React from 'react';
import Header from './Header';

describe('Header component', () => {
  it('renders', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
