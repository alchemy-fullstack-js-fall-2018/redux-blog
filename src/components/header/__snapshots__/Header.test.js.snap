// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders 1`] = `
<div
  id="header"
>
  <h1>
    BLOGGY McBLOG BLOG!
  </h1>
  <nav>
    <Link
      replace={false}
      to="/"
    >
      Home
    </Link>
    <Link
      replace={false}
      to="/users"
    >
      Users
    </Link>
  </nav>
</div>
`;
